### DISCUSSION TITLE
How to generate using Solid's createResource?

### DISCUSSION THREAD

**Discussion by stanlrt** [orval-labs/orval#1742] [General]
How should I configure Orval so it uses createResource instead of Axios, etc... ?

**Comment by melloware** [comment]
I am not sure this is possible right now only Axios and native Fetch client are available as HTTP transports as far as I know.  @soartec-lab I guess he could create a custom HTTP client?

**Reply by soartec-lab** [reply]
Yes. As you said, there are two HTTP clients: `Axios` and `Fetch API`.

@stanlrt 

If you want to automatically generate resource acquisition processing in `SolidJS`, you will need to add a new package as a `client` like `Angular`.

https://github.com/orval-labs/orval/tree/master/packages/angular

**Reply by stanlrt** [reply]
Thanks! 
Seems like someone already did the work, so I ended up using OTQS https://github.com/AndrejNemec/openapi-tanstack-query-solid
